fsw/src/sample_app.c

```diff
@@ -1,461 +1,608 @@
-/*                                                                            */
-/* Name:  SAMPLE_VerifyCmdLength()                                            */
-/*                                                                            */
-/* Purpose:                                                                   */
-/*        Verify command packet length                                        */
-/*                                                                            */
-/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * **/
-static bool SAMPLE_VerifyCmdLength(CFE_SB_MsgPtr_t Msg, uint16 ExpectedLength)
+/*
+** Enable Component
+** TODO: Edit for your specific component implementation
+*/
+void SAMPLE_Enable(void)
 {
-    bool result = true;
-
-    uint16 ActualLength = CFE_SB_GetTotalMsgLength(Msg);
+    int32 status = OS_SUCCESS;
 
-    /*
-    ** Verify the command packet length.
-    */
-    if (ExpectedLength != ActualLength)
+    /* Check that device is disabled */
+    if (SAMPLE_AppData.HkTelemetryPkt.DeviceEnabled == SAMPLE_DEVICE_DISABLED)
     {
-        CFE_SB_MsgId_t MessageID   = CFE_SB_GetMsgId(Msg);
-        uint16         CommandCode = CFE_SB_GetCmdCode(Msg);
+        /* Open device specific protocols */
+        status = uart_init_port(&SAMPLE_AppData.SampleUart);
+        if (status == OS_SUCCESS)
+        {
+            SAMPLE_AppData.HkTelemetryPkt.DeviceCount++;
+            SAMPLE_AppData.HkTelemetryPkt.DeviceEnabled = SAMPLE_DEVICE_ENABLED;
+            CFE_EVS_SendEvent(SAMPLE_ENABLE_INF_EID, CFE_EVS_INFORMATION, "SAMPLE: Device enabled");
+        }
+        else
+        {
+            SAMPLE_AppData.HkTelemetryPkt.DeviceErrorCount++;
+            CFE_EVS_SendEvent(SAMPLE_UART_INIT_ERR_EID, CFE_EVS_ERROR, "SAMPLE: UART port initialization error %d", status);
+        }
+    }
+    else
+    {
+        SAMPLE_AppData.HkTelemetryPkt.DeviceErrorCount++;
+        CFE_EVS_SendEvent(SAMPLE_ENABLE_ERR_EID, CFE_EVS_ERROR, "SAMPLE: Device enable failed, already enabled");
+    }
+    return;
+}
 
-        CFE_EVS_SendEvent(SAMPLE_LEN_ERR_EID, CFE_EVS_EventType_ERROR,
-                          "Invalid Msg length: ID = 0x%X,  CC = %d, Len = %d, Expected = %d",
-                          (unsigned int)CFE_SB_MsgIdToValue(MessageID), CommandCode, ActualLength, ExpectedLength);
 
-        result = false;
+/*
+** Disable Component
+** TODO: Edit for your specific component implementation
+*/
+void SAMPLE_Disable(void)
+{
+    int32 status = OS_SUCCESS;
+
+    /* Check that device is enabled */
+    if (SAMPLE_AppData.HkTelemetryPkt.DeviceEnabled == SAMPLE_DEVICE_ENABLED)
+    {
+        /* Open device specific protocols */
+        status = uart_close_port(SAMPLE_AppData.SampleUart.handle);
+        if (status == OS_SUCCESS)
+        {
+            SAMPLE_AppData.HkTelemetryPkt.DeviceCount++;
+            SAMPLE_AppData.HkTelemetryPkt.DeviceEnabled = SAMPLE_DEVICE_ENABLED;
```

Comment:
Copy paste issue identified by @msuder - Should be `SAMPLE_DEVICE_DISABLED`

Reply:
```
EVS Port1 42/1/SAMPLE 16: SAMPLE: Disable command received
2022-12-14 19:02:02.921647 [ ERROR ] NosEngine.Common - receive operator failed to receive message, connection was shutdown in NosEngine::Transport::Connection::process_stop(std::unique_lock<std::mutex> &) at transport/src/Connection.cpp:377
2022-12-14 19:02:02.921813 [ ERROR ] NosEngine.Common - receive operator caught exception while receiving message, will stop receiving, message connection was shutdown, before async receive message started
EVS Port1 42/1/SAMPLE 17: SAMPLE: Device disabled
```

Reply:
Resolved in commit https://github.com/nasa-itc/sample/pull/1/commits/0474cb32ec8a8612c5c09ae6ddf97b652a7b98f4